apps/server/src/modules/migration/service/migration.service.ts

```diff
@@ -0,0 +1,82 @@
+import { SystemRepo } from '@shared/repo';
+import { Configuration } from '@hpi-schul-cloud/commons/lib';
+import { OauthConfig, System } from '@shared/domain';
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { EntityNotFoundError } from '@shared/common';
+import { PageTypes } from '../controller/dto/page-type.query.param';
+import { PageContentDto } from './dto/page-content.dto';
+
+@Injectable()
+export class MigrationService {
+	private readonly PROCESS_MIGRATION_BASE_URL: string = `${
+		Configuration.get('HOST') as string
+	}/api/v3/migration/process`;
+
+	constructor(private readonly systemRepo: SystemRepo) {}
+
+	async getPageContent(pageType: PageTypes, sourceId: string, targetId: string): Promise<PageContentDto> {
+		const content: PageContentDto = new PageContentDto();
+		const sourceSystem: System = await this.systemRepo.findById(sourceId);
+		const targetSystem: System = await this.systemRepo.findById(targetId);
+
+		content.proceedButtonUrl = this.getLoginUrl(targetSystem, `${this.PROCESS_MIGRATION_BASE_URL}`);
+
+		switch (pageType) {
+			case PageTypes.START_FROM_NEW_SYSTEM:
+				content.proceedButtonUrl = this.getLoginUrl(
+					sourceSystem,
+					this.getLoginUrl(targetSystem, `${this.PROCESS_MIGRATION_BASE_URL}`)
+				);
+				content.cancelButtonUrl = this.getLogoutUrl();
```

Comment:
this should redirect to the login screen, since we are not logged in when coming from the target system